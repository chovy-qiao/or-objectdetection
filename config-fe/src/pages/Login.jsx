import React from "react";
import { connect } from 'react-redux';
import {
    Button,
    TextField,
    Grid,
    Paper,
    Typography,
    Link,
    Container
} from "@material-ui/core";
import { withStyles } from "@material-ui/core/styles";
import { withRouter } from 'react-router-dom';

import { login } from "../store/modules/auth/authSlice";


const useStyles = (theme) => ({
    loginform: {
        justifycontent: "center",
        height: "75%",
    },
    buttonblock: {
        width: "100%",
    },
    loginbackground: {
        justifycontent: "center",
        minheight: "30vh",
        padding: 50,
        width: 400,
        height: 500,
    },
    title1: {
        marginBottom: 40,
    },
    container: {
        top: "40%",
        marginTop: "10%",
    },
});

class Login extends React.Component {
    constructor(props) {
        super(props);
        this.state = { username: "", password: "", authflag: 1 };
        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(event) {
        this.setState({
            username: event.state.username,
            password: event.state.password,
        });
    }
    handleSubmit(event) {
        event.preventDefault()
        this.props.dispatch(login(this.state.username, this.state.password, this.props.history))
    }

    render() {
        const { classes } = this.props;
        return (
            <div>

                <Typography component="h6" variant="h6" align="center">
                    ID: admin@or.com WW: admin
                </Typography>
                <Grid
                    container
                    className={classes.container}
                    spacing={0}
                    justify="center"
                    direction="row"
                >

                    <Grid item>
                        <Grid
                            container
                            direction="column"
                            justify="center"
                            spacing={2}
                            className={classes.loginform}
                            paddingTop={100}
                        >
                            <Paper
                                variant="elevation"
                                elevation={2}
                                className={classes.loginbackground}
                            >
                                <Grid item className={classes.title1}>
                                    <Typography component="h1" variant="h4" align="center">
                                        Login
                                    </Typography>
                                </Grid>
                                <Grid item>
                                    <form onSubmit={this.handleSubmit}>
                                        <Grid container direction="column" spacing={2}>
                                            <Grid item>
                                                <TextField
                                                    type="email"
                                                    placeholder="Email"
                                                    fullWidth
                                                    name="username"
                                                    variant="outlined"
                                                    value={this.state.username}
                                                    onChange={(event) =>
                                                        this.setState({
                                                            [event.target.name]: event.target.value,
                                                        })
                                                    }
                                                    required
                                                    autoFocus
                                                />
                                            </Grid>
                                            <Grid item>
                                                <TextField
                                                    type="password"
                                                    placeholder="Wachtwoord"
                                                    fullWidth
                                                    name="password"
                                                    variant="outlined"
                                                    value={this.state.password}
                                                    onChange={(event) =>
                                                        this.setState({
                                                            [event.target.name]: event.target.value,
                                                        })
                                                    }
                                                    required
                                                />
                                            </Grid>
                                            <Grid item>
                                                <Button
                                                    variant="contained"
                                                    color="primary"
                                                    type="submit"
                                                    className={classes.buttonblock}
                                                >
                                                    Inloggen
                                                </Button>
                                            </Grid>
                                        </Grid>
                                    </form>
                                </Grid>
                                <Grid item>
                                    <Link href="#" variant="body2">
                                        Wachtwoord vergeten?
                                    </Link>
                                </Grid>

                            </Paper>

                        </Grid>

                    </Grid>

                </Grid>

            </div>
        );
    }
}
export default withRouter(connect()(withStyles(useStyles)(Login)))
