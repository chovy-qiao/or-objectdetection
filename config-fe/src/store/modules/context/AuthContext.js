import React from "react";

export const AuthenticationContext = React.createContext();
export const useAuthenticationContext = () =>
  React.useContext(AuthenticationContext);
/* HOC to inject store to any functional or class component */
export const withAuthenticationContext = (Component) => (props) => {
  return (
    <Component {...props} authenticationService={useAuthenticationContext()} />
  );
};
export default AuthenticationContext;
