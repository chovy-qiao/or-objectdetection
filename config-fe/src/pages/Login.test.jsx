import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Login from "./Login";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import { AuthenticationContext } from "../store/modules/context/AuthContext";

const renderLogin = ({
  authService = { login: jest.fn() },
  location = { state: { username: "harry" } }
} = {}) => {
  return render((store) => (
    <MemoryRouter>
      <AuthenticationContext.Provider value={authService}>
        <Login location={location} store={store} />
      </AuthenticationContext.Provider>
    </MemoryRouter>
  ));
};

describe("LoginScreen", () => {
  it("Renders correctly", () => {
    const { container } = renderLogin();
    expect(container).toMatchSnapshot();
  });
});
