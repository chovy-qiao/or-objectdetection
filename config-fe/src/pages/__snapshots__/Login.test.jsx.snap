// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen Renders correctly 1`] = `<div />`;
